src/Microsoft.TemplateEngine.Core/Util/ProcessorState.cs

```diff
@@ -75,7 +75,8 @@ public ProcessorState(Stream source, Stream target, int bufferSize, int flushThr
             CurrentBufferLength = source.Read(CurrentBuffer, 0, CurrentBuffer.Length);
 
             Encoding encoding = EncodingUtil.Detect(CurrentBuffer, CurrentBufferLength, out _bom);
-            Encoding = encoding;
+            _encoding = encoding;
```

Comment:
not sure of this change - why it was needed?

Reply:
It's to make compiler happy about #nullable enable so this 2 fields are set in constructor, wondering what would be better way to do this...

Reply:
Looks like there is much better solution, but only in C#9 :(
Where you can put attribute on method that is called from constructor and it will mark field as set...

<img width="951" alt="image" src="https://user-images.githubusercontent.com/774791/111435249-d51b0680-8700-11eb-9c1e-40e61ef0b079.png">

Here is sharplab gist you can play with feature: https://sharplab.io/#gist:17b51a72b4f952b7ef06566c7e5208f1

What would you prefer to do now?

Reply:
I see that setter is not used at all after this change. Should we remove it and make both properties read-only?